models/solution/stop.rb

```diff
@@ -94,6 +94,21 @@ def build_skills
         skills_to_output += self.original_skills
         skills_to_output
       end
+
+      def active_timewindow
+        return nil if activity.timewindows.empty?
+
+        if info.waiting_time > 0
+          active_tw = activity.timewindows.select{ |tw| tw.start > info.begin_time }.min_by{ |tw|
+            tw.start - info.begin_time
+          }
+          return active_tw
+        end
+
+        activity.timewindows.find{ |tw|
+          tw.start <= info.begin_time && info.begin_time <= tw.safe_end(activity.lateness_allowed?)
+        }
```

Comment:
It is better to check first without "lateness" first (and then check it with lateness only if the first doesn't find a TW) because with lateness activated, time windows might overlap and we can identify an incorrect time window.